### PR TITLE
Stable devcrypto init [1.1.1]

### PR DESCRIPTION
**Pull request by cotequeiroz**
A couple of cherry-picks from #8213 to OpenSSL_1_1_1_stable

### CODE REVIEW

File: crypto/engine/eng_devcrypto.c [hunk 1]
```diff
@@ -284,15 +297,29 @@ static int ctr_do_cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
...
 {
+    struct cipher_ctx *cipher_ctx =
+        (struct cipher_ctx *)EVP_CIPHER_CTX_get_cipher_data(ctx);
     EVP_CIPHER_CTX *to_ctx = (EVP_CIPHER_CTX *)p2;
-    struct cipher_ctx *cipher_ctx;
+    struct cipher_ctx *to_cipher_ctx;
 
-    if (type == EVP_CTRL_COPY) {
+    switch (type) {
+
```
Comment (romen) [inline crypto/engine/eng_devcrypto.c]: nit: should we remove the empty line between the beginning of the `switch` statement and the first `case`?

Comment (cotequeiroz) [reply crypto/engine/eng_devcrypto.c]: I can surely do that.  Is it OK if it differs from master?

**Comment by romen** [comment]
Removed the `ready` label as force-push requires explicit reapproval from @levitte

**Comment by mattcaswell** [comment]
Pushed.

**Pull request closed without merging**